Panic: nil pointer dereference in SecretsManagerSecrets.deleteAsync when DescribeSecret fails (v0.45.0/v0.46.0)

Hey! We’re hitting a hard crash when nuking AWS Secrets Manager secrets. Cloud-nuke panics with a nil pointer dereference in SecretsManagerSecrets.deleteAsync:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0x6fbb3c8]

goroutine 1240 [running]:
github.com/gruntwork-io/cloud-nuke/aws/resources.(*SecretsManagerSecrets).deleteAsync(...)
  /home/circleci/project/aws/resources/secrets_manager.go:103 +0xe8
created by github.com/gruntwork-io/cloud-nuke/aws/resources.(*SecretsManagerSecrets).nukeAll
  /home/circleci/project/aws/resources/secrets_manager.go:76 +0x14d
```

**Versions**

- cloud-nuke v0.45.0 (repro)
- Also appears to still be present in v0.46.0 (same code path)

**What happens**

From reading the code, `deleteAsync` calls `DescribeSecret(...)` but does not check err before using the returned secret value. It then does:

- `len(secret.ReplicationStatus) > 0`

If `DescribeSecret` returns an error (throttling / access denied / not found due to race / transient API error), secret can be `nil`, which causes `SIGSEGV`.

**Expected behavior**

Cloud-nuke should not crash. If `DescribeSecret` fails, the error should be returned/logged for that resource and the run should continue.

**Suggested fix (minimal)**

In `aws/resources/secrets_manager.go` inside `deleteAsync`, add a guard right after `DescribeSecret`:

- `if err != nil` → send to `errChan` and return
- `if secret == nil` → send an error to `errChan` and return

--- a/aws/resources/cloudtrail.go
+++ b/aws/resources/cloudtrail.go
@@ -54,6 +54,9 @@ func listCloudtrailTrails(ctx context.Context, client CloudtrailTrailAPI, scope 
 				ResourceIdList: []string{*trail.TrailARN},
 			}); err == nil && len(tags.ResourceTagList) > 0 {
 				for _, tag := range tags.ResourceTagList[0].TagsList {
+					if tag.Key == nil || tag.Value == nil {
+						continue
+					}
 					rv.Tags[*tag.Key] = *tag.Value
 				}
 			}

--- a/aws/resources/secrets_manager.go
+++ b/aws/resources/secrets_manager.go
@@ -2,6 +2,7 @@ package resources
 
 import (
 	"context"
+	"fmt"
 	"time"
 
 	"github.com/aws/aws-sdk-go-v2/aws"
@@ -92,6 +93,9 @@ func deleteSecretsManagerSecret(ctx context.Context, client SecretsManagerAPI, s
 	})
 	if err != nil {
 		return err
+	}
+	if secret == nil {
+		return fmt.Errorf("DescribeSecret returned nil for secret %s", aws.ToString(secretID))
 	}
 
 	// Delete replications if this is a primary secret with replicas

--- a/aws/resources/secrets_manager_test.go
+++ b/aws/resources/secrets_manager_test.go
@@ -2,6 +2,7 @@ package resources
 
 import (
 	"context"
+	"fmt"
 	"regexp"
 	"testing"
 	"time"
@@ -17,6 +18,7 @@ import (
 type mockSecretsManagerClient struct {
 	ListSecretsOutput                  secretsmanager.ListSecretsOutput
 	DescribeSecretOutput               secretsmanager.DescribeSecretOutput
+	DescribeSecretError                error
 	DeleteSecretOutput                 secretsmanager.DeleteSecretOutput
 	RemoveRegionsFromReplicationOutput secretsmanager.RemoveRegionsFromReplicationOutput
 }
@@ -26,6 +28,9 @@ func (m *mockSecretsManagerClient) ListSecrets(ctx context.Context, params *secr
 }
 
 func (m *mockSecretsManagerClient) DescribeSecret(ctx context.Context, params *secretsmanager.DescribeSecretInput, optFns ...func(*secretsmanager.Options)) (*secretsmanager.DescribeSecretOutput, error) {
+	if m.DescribeSecretError != nil {
+		return nil, m.DescribeSecretError
+	}
 	return &m.DescribeSecretOutput, nil
 }
 
@@ -116,4 +121,16 @@ func TestDeleteSecretsManagerSecret_WithReplicas(t *testing.T) {
 
 	err := deleteSecretsManagerSecret(context.Background(), mock, aws.String("arn:aws:secretsmanager:us-east-1:123456789012:secret:test-secret"))
 	require.NoError(t, err)
+}
+
+func TestDeleteSecretsManagerSecret_DescribeSecretError(t *testing.T) {
+	t.Parallel()
+
+	mock := &mockSecretsManagerClient{
+		DescribeSecretError: fmt.Errorf("throttling: rate exceeded"),
+	}
+
+	err := deleteSecretsManagerSecret(context.Background(), mock, aws.String("arn:aws:secretsmanager:us-east-1:123456789012:secret:test-secret"))
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "throttling")
 }

